core/src/replay_stage.rs

```diff
@@ -158,28 +144,9 @@ impl ReplayStage {
                             .lock()
                             .unwrap()
                             .reset(parent.tick_height(), parent.last_blockhash());
-
-                        if next_leader == my_id {
-                    		let frozen = bank_forks.read().unwrap().frozen_banks();
-							assert!(frozen.get(&next_slot).is_none());
-                            assert!(bank_forks.read().unwrap().get(next_slot).is_none());
-
-                            let tpu_bank = Bank::new_from_parent(&parent, my_id, next_slot);
-                            bank_forks.write().unwrap().insert(next_slot, tpu_bank);
-                            if let Some(tpu_bank) =
-                                bank_forks.read().unwrap().get(next_slot).cloned()
-                            {
-								assert_eq!(bank_forks.read().unwrap().working_bank().slot(), tpu_bank.slot());
-                                debug!(
-                                    "poh_recorder new working bank: me: {} next_slot: {} next_leader: {}",
-                                    my_id,
-                                    tpu_bank.slot(),
-                                    next_leader
-                                );
-                				poh_recorder.lock().unwrap().set_bank(&tpu_bank);
-                            }
-                        }
                     }
+
+                    Self::start_leader(my_id, &bank_forks, &poh_recorder, &cluster_info);
```

Comment:
if the replay_blocktree_into_bank()  processing above takes a long time, can the poh_recorder potentially tick past the slot for which this node is supposed to be the leader?

Reply:
Yea. Unless that block completes and the poh is reset